papyrus/geo_interface.py

```diff
@@ -100,10 +109,14 @@ def __update__(self, feature):
                     self._shape = shape
             elif not col.primary_key:
                 if p.key in feature.properties:
+                    if col.info.get('readonly'):
+                        continue
```

Comment:
I don't understand why this test does not come earlier, right after `col = p.columns[0]`? And what about geometry columns? Don't we want to support readonly geometry columns as well?

Reply:
Here we need to check `p.key in feature.properties` or `geom and not isinstance(geom, geojson.geometry.Default` before raising an exception.
I just added the same behavior on geometry columns.